### PR TITLE
MM-10699 Add channel renaming to CLI

### PR DESCRIPTION
**Pull request by theblueskies**

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -159,6 +159,27 @@ func (a *App) CreateChannelWithUser(channel *model.Channel, userId string) (*mod
 	return rchannel, nil
 }
 
+// RenameChannel is used to rename the channel Name and the DisplayName fields
+func (a *App) RenameChannel(channel *model.Channel, newChannelName string, newDisplayName string) (*model.Channel, *model.AppError) {
+	var oldChannel *model.Channel
```
Comment (jespino) [inline app/channel.go]: You are using 2 different styles in the same function. We prefer the later, something like:
```go
oldChannel, err := a.GetChannel(channel.Id)
if err != nil {
	return nil, err
}
```

File: cmd/mattermost/commands/channel_test.go [hunk 2]
```diff
@@ -113,3 +113,11 @@ func TestCreateChannel(t *testing.T) {
 	name = name + "-private"
 	CheckCommand(t, "channel", "create", "--display_name", name, "--team", th.BasicTeam.Name, "--private", "--name", name)
 }
+
+func TestRenameChannel(t *testing.T) {
+	th := api4.Setup().InitBasic()
+	defer th.TearDown()
+
+	channel := th.CreatePublicChannel()
+	CheckCommand(t, "channel", "rename", th.BasicTeam.Name+":"+channel.Name, "NewChannelName", "--display_name", "New Display Name")
```
Comment (jespino) [inline cmd/mattermost/commands/channel_test.go]: I would add here the assertions about the channel object renamed in the database too.

File: cmd/mattermost/commands/channel_test.go [hunk 3]
```diff
@@ -113,3 +114,17 @@ func TestCreateChannel(t *testing.T) {
...
 	CheckCommand(t, "channel", "create", "--display_name", name, "--team", th.BasicTeam.Name, "--private", "--name", name)
 }
+
+func TestRenameChannel(t *testing.T) {
+	th := api4.Setup().InitBasic()
+	defer th.TearDown()
+
+	channel := th.CreatePublicChannel()
+	CheckCommand(t, "channel", "rename", th.BasicTeam.Name+":"+channel.Name, "newchannelname10", "--display_name", "New Display Name")
+	RunCommand(t, "channel", "rename", th.BasicTeam.Name+":"+"channel.Name", "newchannelname10", "--display_name", "New Display Name")
```
Comment (jespino) [inline cmd/mattermost/commands/channel_test.go]: I think this second line (`RunCommand` one) is not necessary, actually, I doesn't check anything.

Comment (theblueskies) [reply cmd/mattermost/commands/channel_test.go]: You're right!

**Comment by grundleborg** [comment]
@theblueskies please could you rebase this onto the latest `master` and then force push to the branch? Then we can get it merged in.

**Comment by jespino** [comment]
@theblueskies We found [this bug](https://mattermost.atlassian.net/browse/MM-12165) reviewing the channel rename feature. Actually, it's not a bug in your code, is a bug in the already existing rename code, but maybe you are interested in solve this too. If you want, let me know, if not, we can take care of this :)

**Pull request merged**